plugins/modules/ec2_vpc_nat_gateway.py

```diff
@@ -632,30 +577,30 @@ def release_address(client, allocation_id, check_mode=False):
     Returns:
         Boolean, string
     """
-    err_msg = ''
-    if check_mode:
+    msg = ''
+
+    if module.check_mode:
         return True, ''
 
     ip_released = False
     try:
         client.describe_addresses(aws_retry=True, AllocationIds=[allocation_id])
-    except botocore.exceptions.ClientError as e:
+    except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
```

Comment:
There might be a more specific failure that we could catch here with is_boto3_error_code (probably InvalidAllocationID.NotFound, not sure about any others) to do the `return True, str(e)`, and handle general exceptions separately.

Reply:
@jillr Thank you for reviewing. I tried to add the catch on this error. Let me know if it makes sense to you.

Reply:
It should be possible to catch both BotoCoreError and ClientError.

While pylint thinks that `except is_boto3_error_code('InvalidAllocationID.NotFound')` catches all ClientError, this is actually incorrect and due to a bug in pylint.  https://github.com/PyCQA/pylint/issues/2174

Reply:
@tremble Ok, my Shippable check was failing due to this duplicate catch. I added again the the `botocore.exceptions.ClientError`. Thanks.